src/cuda_derivs.h

```diff
@@ -35,12 +35,10 @@ __device__ __forceinline__ __attribute__((always_inline)) void fluxQuadSharedx(m
 	flxm = 0.0;
 	__syncthreads();
 
-	for (int lt=1; lt<stencilSize+1; lt++)
-		for (int mt=0; mt<lt; mt++) {
-			flxp -= dcoeffF[stencilSize-lt]*(s_f[si-mt]+s_f[si-mt+lt])*(s_g[si-mt]+s_g[si-mt+lt]);
-			flxm -= dcoeffF[stencilSize-lt]*(s_f[si-mt-1]+s_f[si-mt+lt-1])*(s_g[si-mt-1]+s_g[si-mt+lt-1]);
-		}
-
+	for (int lt=1; lt<stencilSize+1; lt++) {
+                        flxp -= dcoeffF[stencilSize-lt]*(s_f[si]+s_f[si+lt])*(s_g[si]+s_g[si+lt]);
+			flxm -= dcoeffF[stencilSize-lt]*(s_f[si]+s_f[si-lt])*(s_g[si]+s_g[si-lt]);
+        }
```

Comment:
why this change?

Reply:
It is as per the paper of Kuya and Kawai (2021). Basically they said that two loops are not needed to compute the flux. Same thing can be computed in one loop as a lot of terms cancel when the difference is taken.